src/types/smartSearch.ts

```diff
@@ -18,3 +18,39 @@ export enum TIME_FRAME {
     BETWEEN='between',
     LAST_FEW_DAYS='lastFew',
 }
+
+/**
+ * Filter Configs
+ */
+export type FilterConfig = Record<string, unknown> // Filter Configs are objects
+
+export type DefaultFilterConfig = Record<string, never> // Default filter config is an empty object
+
+export interface MostActiveFilterConfig extends FilterConfig {
+    after?: string;
+    before?: string;
+    size?: number;
+}
+
+export type AnyFilterConfig = DefaultFilterConfig | MostActiveFilterConfig // Add all filter objects here
+
+/** Filters */
+export interface ZetkinSmartSearchFilter<C extends FilterConfig = AnyFilterConfig> {
+    config?: C;
+    op: OPERATION;
+    type: FILTER_TYPE;
+}
+
+export interface SmartSearchFilterWithId<C extends FilterConfig = AnyFilterConfig> extends ZetkinSmartSearchFilter<C> {
+    id: number;
+}
+
+export interface NewSmartSearchFilter {
+    type: FILTER_TYPE;
+}
```

Comment:
Don't understand why we need this type? Will there ever be a new filter? When creating a new filter, they come with a default config and op, up to the user to change it

Reply:
Hopefully answer below explains this